examples/goldilocks_models/find_goldilocks_models.cc

```diff
@@ -1664,20 +1717,41 @@ int findGoldilocksModels(int argc, char* argv[]) {
   }
   const int method_to_solve_system_of_equations =
       FLAGS_method_to_solve_system_of_equations;
-  // With bigger momentum, you might need a larger tolerance
+  // With bigger momentum, you might need a larger tolerance.
+  // If not uniform grid, considering sample cost increase makes no sense.
+  // So, use a very large tolerance on increase rate.
   double max_sample_cost_increase_rate = 0;
   if (FLAGS_robot_option == 0) {
-    max_sample_cost_increase_rate = FLAGS_is_stochastic? 2.0: 0.01;
+      if(uniform_grid){
+          max_sample_cost_increase_rate = FLAGS_is_stochastic? 2.0: 0.01;
+      }else{
+          max_sample_cost_increase_rate = 100;
```

Comment:
You can set this to be `std::numeric_limits<double>::infinity()`, since we are not using it if I understand your code correctly

Reply:
Resolved